frontstage/views/contact_us.py

```diff
@@ -60,21 +63,55 @@ def send_message(session) -> str:
         except InvalidSecureMessagingForm as e:
             errors = _errors(e.errors)
 
-    respondent_enrolments = get_respondent_enrolments(session.get_party_id())
+    url_parts = urlparse.urlparse(request.url)
+    url_business_id = urlparse.parse_qs(url_parts.query).get("business_id")
+    payload = {"business_id": url_business_id[0]} if url_business_id else {}
 
-    if len(respondent_enrolments) > 2:
-        return redirect(url_for("surveys_bp.get_survey_list", tag="todo"))
+    respondent_enrolments = get_respondent_enrolments(session.get_party_id(), payload=payload)
+
+    if len(respondent_enrolments) > 1:
+        return redirect(url_for("contact_us_bp.select_organisation"))
 
     enrolment_options = secure_message_enrolment_options(respondent_enrolments[0], secure_message_form)
+    back_url = (
+        url_for("contact_us_bp.select_organisation")
+        if url_business_id
+        else url_for("surveys_bp.get_survey_list", tag="todo")
+    )
 
     return render_template(
         "secure-messages/help/secure-message-send-messages-view.html",
         enrolment_options=enrolment_options,
+        back_url=back_url,
         form=secure_message_form,
         errors=errors,
     )
 
 
+@contact_us_bp.route("/select-organisation", methods=["GET", "POST"])
+@jwt_authorization(request)
+def select_organisation(session) -> str:
+    organisation_form = OrganisationForm(request.form)
+
+    if request.method == "POST" and organisation_form.validate():
+        return redirect(url_for("contact_us_bp.send_message", business_id=organisation_form.business_id.data))
+
+    respondent_enrolments = get_respondent_enrolments(session.get_party_id())
```

Comment:
You could argue this is slightly inefficient as there is a 2nd call to get_respondent_enrolments on the journey. i.e go to send_message redirects to this page if more than one business that calls get_respondent_enrolments again. However as this is a niche journey I think it is far cleaner! and easier to follow doing this. Trying to do it all in one endpoint has all sorts of issues with validation and keeping state, added on erroring, this is realistically the only solution. Could I pass the list of businesses from send_message redirect, yes, but I think that gets messy in the url or session and also makes the select organisation page very much dependent.